### DISCUSSION TITLE
Cropping videos to improve accuracy of predictions

### DISCUSSION THREAD

**Discussion by clee758** [talmolab/sleap#1388] [General]
Hi,

I was told that if the wells of each video are not in the exact same position, this will affect the predicting accuracy of SLEAP. Would you recommend cropping the videos to make the well position more uniform among videos? Or is there another solution?

**Comment by talmo** [comment]
Hi @clee758,

Wherever possible, any form of standardization will make things easier for machine learning models since it's one less source of variability to deal with.

If it's easy enough to do it manually, I'd recommend it since it'll reduce the chances of mistakes happening (e.g., tracks jumping across wells) and will speed up both training and inference. For bottom-up in partciular, this might make it more tractable as processing the crops of the wells will require a lot less memory than processing the whole image.

If they're not in the same location in every video, you might consider training a lightweight SLEAP model to detect the corners of the wells. You can run this model on a handful of frames per video (no need to do it on every frame since they should be static), and use that info to crop the wells out.

If that's too complicated, it's ok to do it without it. A top-down model will probably work best in terms of efficiency.

Let us know if you have any questions!

Cheers,

Talmo

**Reply by clee758** [reply]
Hi, thank you so much! I was wondering how you would use the lightweight SLEAP model to detect the corners of the well? Thank you!

**Reply by roomrys** [reply]
Hi @clee758,

To use a SLEAP model to detect the corners of the well, you would create a new project with a skeleton specific to the object you want to crop to. Since the object is (relatively) stationary, you should be able to get good prediction by just labeling a few frames, try labeling 10 frames and check the predictions. The model should give you the exact pixel coordinates to crop to in each video. The advantage of this method would be precision and standardization of the crops.

Thanks,
Liezl

**Reply by clee758** [reply]
Is there a way to import labels from an original video onto a cropped version?

**Reply by talmo** [reply]
Not super easily unfortunately :(

If you're comfortable with Python, you can use the SLEAP API to adjust the coordinates and save out an adjusted set of labels for the cropped version, but the way you'd go about it is case-specific.

If you have a Colab with example labels that you want cropped, and where the cropping information is coming from, we'd be happy to help with a custom script to do this though. Just let us know!

**Comment by clee758** [comment]
Hi, I cropped the movies, but the cropped movies are not centered in the same place in SLEAP. Is that going to affect the prediction?

**Reply by talmo** [reply]
Nope! As long as you train on those movies it should work just fine and learn to ignore variability in the exact centering.

Let us know if that's not the case though!

Cheers,

Talmo
